src/core/loadbytecode.c

```diff
@@ -7,6 +7,43 @@ static void run_load(MVMThreadContext *tc, void *sr_data);
 static void mark_sr_data(MVMThreadContext *tc, MVMFrame *frame, MVMGCWorklist *worklist) {
     MVM_gc_worklist_add(tc, worklist, &frame->special_return_data);
 }
+void MVM_run_comp_unit(MVMThreadContext *tc, MVMCompUnit *cu) {
+    /* If there's a deserialization frame, need to run that. */
+    if (cu->body.deserialize_frame) {
+        /* Set up special return to delegate to running the load frame,
+         * if any. */
+        tc->cur_frame->return_value             = NULL;
+        tc->cur_frame->return_type              = MVM_RETURN_VOID;
+        tc->cur_frame->special_return           = run_load;
+        tc->cur_frame->special_return_data      = cu;
+        tc->cur_frame->mark_special_return_data = mark_sr_data;
+
+        /* Invoke the deserialization frame and return to the runloop. */
+        MVM_frame_invoke(tc, cu->body.deserialize_frame, MVM_callsite_get_common(tc, MVM_CALLSITE_ID_NULL_ARGS),
+            NULL, NULL, NULL, -1);
+    }
+    else {
+        /* No deserialize frame, so do load frame instead. */
+        run_load(tc, cu);
+    }
+}
+void MVM_load_bytecode_buffer(MVMThreadContext *tc, MVMObject *buf) {
+    MVMCompUnit *cu;
+    MVMuint8    *data_start;
+    MVMuint32    data_size;
+
+    /* Ensure the source is in the correct form. */
+    if (!IS_CONCRETE(buf) || REPR(buf)->ID != MVM_REPR_ID_MVMArray)
+        MVM_exception_throw_adhoc(tc, "loadbytecodebuffer requires a native array to read from");
+
+    /* MVMCompUnit expects the data to be non-GC managed as it usually comes straight from a file */
+    data_size = ((MVMArray *)buf)->body.elems;
+    data_start = MVM_malloc(data_size);
+    memcpy(data_start, (MVMuint8 *)(((MVMArray *)buf)->body.slots.i8 + ((MVMArray *)buf)->body.start), data_size);
```

Comment:
Should also check that the the buffer's element type is 8-bit integers. It could easily be something else.